api/v1beta1/ssp_types.go

```diff
@@ -56,13 +56,13 @@ type NodeLabeller struct {
 // SSPSpec defines the desired state of SSP
 type SSPSpec struct {
 	// TemplateValidator is configuration of the template validator operand
-	TemplateValidator TemplateValidator `json:"templateValidator,omitempty"`
+	TemplateValidator *TemplateValidator `json:"templateValidator,omitempty"`
```

Comment:
Can you add `nil` check to places where this is used?
For example here:
https://github.com/kubevirt/ssp-operator/blob/e67e36faecec9a3a45f290219e2bc2e297958870/internal/operands/template-validator/reconcile.go#L149-L154

And many places in functional tests.